internal/handlers/CreateUser.go

```diff
@@ -41,9 +41,19 @@ func CreateUser(w http.ResponseWriter, r *http.Request) {
 			UserName: userResponse.UserName,
 			Password: passwordHashed,
 		}
-		rep.SaveUser(user)
+		err = rep.SaveUser(user)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
add please here some log, for example fmt.Println(err) because  if this error happened you will never understand why